superscore/widgets/snapshot_mainpage.py

```diff
@@ -0,0 +1,85 @@
+'''
+Currently opens "main page" that contians the snapshot view. This is the main page that is mocked up on MIRO with basic
+button to dialog screen opening functionaliy. Including a skeleton for page functionality.
+'''
+from qtpy import QtWidgets
+
+from superscore.widgets.core import Display
+from superscore.widgets.filter_page import FilterMenu
+from superscore.widgets.load_snapshot import LoadScreen
+from superscore.widgets.save_screen import SaveScreen
+
+
+class SnapshotMainpage(Display, QtWidgets.QMainWindow):
+    filename = "snapshot_display_mainpage.ui"
+
+    # collectionsVertLayout
+    collSearchButton: QtWidgets.QPushButton
+    searchLineEdit: QtWidgets.QLineEdit
+    collectionsLabel: QtWidgets.QLabel
+    collectionsTreeView: QtWidgets.QTreeView
+
+    # Top box layout load/save/info buttons
+    infoButton: QtWidgets.QPushButton
+    loadConfigButton: QtWidgets.QPushButton
+    saveButton: QtWidgets.QPushButton
+
+    # filters layout
+    chooseMonthSpinBox: QtWidgets.QSpinBox
+    displaySnapshotLabel: QtWidgets.QLabel
+    filterMenuButton: QtWidgets.QPushButton
+    monthsAgoLabel: QtWidgets.QLabel
+    searchDescButton: QtWidgets.QPushButton
+    searchDescLineEdit: QtWidgets.QLineEdit
+    searchTitleLineEdit: QtWidgets.QLineEdit
+    searchTitleButton: QtWidgets.QPushButton
+
+    loadPage: QtWidgets.QWidget
+
+    # snapshot layout
+    snapshotTableWidget: QtWidgets.QTableWidget
+
+    def search_string_title(self):
+        searchTerm = self.searchTitleLineEdit.text()
+        self.refreshTable(searchTerm)
+        pass
+
+    def open_save_page(self):
+        self.savePage = SaveScreen()
+        self.savePage.show()
+
+    def open_info_log(self):
+        pass
+
+    def open_filter_menu(self):
+        self.filter = FilterMenu()
```

Comment:
```suggestion
        self.filter = FilterMenu(parent=self)
        self.filter.setWindowFlag(QtCore.Qt.Dialog)
```

Generally with qt you should give widgets parents, so that if their parent is deleted the child gets cleaned up as well.  This also helps garbage collection not accidentally delete the widget prematurely.  

If you want to preserve the pop-out-window functionality you add the proper `WindowFlag`  (you'll have to import `QtCore.Qt.Dialog` one way or another)